jetty-core/jetty-client/src/main/java/org/eclipse/jetty/client/transport/HttpConversation.java

```diff
@@ -32,115 +35,162 @@ public class HttpConversation extends Attributes.Lazy
     private final Deque<HttpExchange> exchanges = new ConcurrentLinkedDeque<>();
     private volatile ResponseListeners listeners;
 
-    public Deque<HttpExchange> getExchanges()
+    public void offerExchange(HttpExchange exchange)
     {
-        return exchanges;
+        exchanges.offer(exchange);
+    }
+
+    private HttpExchange firstExchange()
+    {
+        HttpExchange exchange = exchanges.peekFirst();
+        assert exchange != null;
+        return exchange;
+    }
+
+    public HttpExchange lastExchange()
+    {
+        HttpExchange exchange = exchanges.peekLast();
+        assert exchange != null;
+        return exchange;
     }
 
     /**
-     * Returns the list of response listeners that needs to be notified of response events.
-     * This list changes as the conversation proceeds, as follows:
+     * <p>Updates the response listeners, eventually using the given override response listener,
+     * that must be notified instead of the response listeners of the first exchange.</p>
```

Comment:
This javadoc doesn't read well.  I can't tell what it is saying?  When is this method called and why?